tests/test_api2.py

```diff
@@ -94,7 +94,7 @@ def test_digita_endppoint_authenticated_access():
     # Replace Time with ~current time
     ts = (datetime.now(timezone.utc) - timedelta(seconds=1)).strftime("%Y-%m-%dT%H:%M:%S.%f%z")
     payload["DevEUI_uplink"]["Time"] = ts
-    resp = httpx.post(url, headers=headers, params=params, data=payload)
+    resp = httpx.post(url, headers=headers, params=params, data=json.dumps(payload))
```

Comment:
It is better to use json=payload when httpx.post()ing, e.g. 

`resp = httpx.post(url, headers=headers, params=params, json=payload)`

Reply:
On my machine dev set up, resp = httpx.post(url, headers=headers, params=params, json=payload) makes the parser error out at json.loads(data["request"]["body"].decode()) because the body comes as url encoded. b'DevEUI_uplink=%7B%27Time%27%3A+%272023-10-17T08%3A32%3A1 .. 
data=json.dumps(payload) fixes this issue. Do parser work with just 'payload' for you ? @aapris